src/export/csv/index.ts

```diff
@@ -0,0 +1,101 @@
+import { GraphQLClient } from "graphql-request";
+import omit from "lodash.omit";
+import { getFlowName, getLatestFlowGraph } from "../../flow";
+import { getSessionById } from "../../session";
+import { getBOPSParams } from "../bops/index";
+import { QuestionAndResponses, Response } from "../bops/model";
+import { CSVData } from "./model";
+
+export async function generateCSVData(
+  client: GraphQLClient,
+  sessionId: string
+): Promise<CSVData> {
+  const session = await getSessionById(client, sessionId);
+  const flow = await getLatestFlowGraph(client, session.flowId);
+  if (!flow)
+    throw new Error(
+      `Cannot get flow ${session.flowId}, therefore cannot generate CSV data.`
+    );
+  const flowName = await getFlowName(client, session.flowId);
+
+  const bopsData = getBOPSParams({
+    breadcrumbs: session.data.breadcrumbs,
+    flow: flow,
+    passport: session.data.passport,
+    sessionId: session.id,
+    flowName: flowName,
+  });
+
+  // format dedicated BOPS properties (eg `applicant_email`) as list of questions & responses to match proposal_details
+  //   omitting debug data and payload keys already in confirmation details
+  const summary: Record<string, any> = {
+    ...omit(bopsData, ["planx_debug_data", "files", "proposal_details"]),
+  };
+  const formattedSummary: { question: string; responses: Array<Response> }[] =
+    [];
+  Object.keys(summary).forEach((key) => {
+    formattedSummary.push({
+      question: key,
+      responses: [{ value: summary[key] }],
+    });
+  });
+
+  // similarly format file uploads as list of questions, responses, metadata
+  const formattedFiles: QuestionAndResponses[] = [];
+  bopsData["files"]?.forEach((file) => {
+    formattedFiles.push({
+      question: file.tags
+        ? `File upload: ${file.tags.join(", ")}`
+        : "File upload",
+      responses: [
+        { value: file.filename.split("/").pop() || "Unknown filename" },
+      ],
+      metadata: { notes: file.applicant_description || "" },
+    });
+  });
+
+  // gather key reference fields, these will be first rows of CSV
+  const references: { question: string; responses: Array<Response> }[] = [
+    {
+      question: "Planning Application Reference", // match language used on Confirmation page
+      responses: [{ value: sessionId }],
+    },
+    {
+      question: "Property Address",
+      responses: [
+        {
+          value: [
+            bopsData.site?.address_1,
+            bopsData.site?.address_2,
+            bopsData.site?.town,
+            bopsData.site?.postcode,
+          ]
+            .filter(Boolean)
+            .join(" ")
+            .replaceAll(",", ""), // omit commas for csv > pdf parsing later by Uniform
```

Comment:
This is for custom solutions right, such as Kev's CSV -> PDF generation for Bucks?

If so, won't any string with commas break this generation, e.g. a project description of "something, something, and something".

It might be better to parse the entire string, escaping strings containing commas with double quotes. There may well be an option for this in the CSV library we're using?

Appreciate this isn't a new issue introduced here but worth flagging and we can raise an issue if this is going to be fixed later 👍

Reply:
This shout about Kev's custom solution is a good one & I just remembered this also while going to update `planx-document-templates`. 

I think it's actually a reason _not_ to change the CSV formatting yet to align all rows to the `Response` type - just because I don't think we have any confirmation that Majad has taken over that workflow and it might result in a slow testing turnaround time / even more processing woes. 

Let's come back to the improving the actual structure of the CSV and removing parsing hacks like this line in a separate PR :+1: I'll keep feature parity with the current CSV for now.